File: Java/ATM.java
import java.util.Scanner;

public class ATM {

    public static String formatRupiah(long angka) {
        return String.format("Rp %,d", angka).replace(",", ".");
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String nama, valid;
        long saldo, jumlah;

        do {
            System.out.print("Masukkan Nama: ");
            nama = sc.nextLine();
            System.out.print("Apakah nama \"" + nama + "\" benar? (TRUE/FALSE): ");
            valid = sc.nextLine().toUpperCase();
        } while (!valid.equals("TRUE"));

        System.out.print("Masukkan NIM tanpa huruf: ");
        saldo = sc.nextLong();

        System.out.println("\nNama: " + nama);
        System.out.println("Saldo Awal: " + formatRupiah(saldo));

        int pilihan;
        while (true) {
            System.out.println("\n===== MENU ATM =====");
            System.out.println("1. Cek Saldo");
            System.out.println("2. Tarik Tunai");
            System.out.println("3. Setor Tunai");
            System.out.println("4. Transfer");
            System.out.println("5. Keluar");
            System.out.print("Pilih menu: ");

            pilihan = sc.nextInt();

            switch (pilihan) {
                case 1:
                    System.out.println("Saldo Anda: " + formatRupiah(saldo));
                    break;

                case 2:
                    System.out.print("Masukkan jumlah tarik: ");
                    jumlah = sc.nextLong();
                    if (jumlah > saldo) {
                        System.out.println("Saldo tidak cukup!");
                    } else {
                        saldo -= jumlah;
                        System.out.println("Saldo sekarang: " + formatRupiah(saldo));
                    }
                    break;

                case 3:
                    System.out.print("Masukkan jumlah setor: ");
                    jumlah = sc.nextLong();
                    saldo += jumlah;
                    System.out.println("Saldo sekarang: " + formatRupiah(saldo));
                    break;

                case 4:
                    System.out.print("Masukkan jumlah transfer: ");
                    jumlah = sc.nextLong();
                    if (jumlah > saldo) {
                        System.out.println("Saldo tidak mencukupi!");
                    } else {
                        saldo -= jumlah;
                        System.out.println("Transfer berhasil. Saldo sekarang: " + formatRupiah(saldo));
                    }
                    break;

                case 5:
                    System.out.println("Terima kasih telah menggunakan ATM!");
                    System.exit(0);

                default:
                    System.out.println("Pilihan tidak valid!");
            }
        }
    }
}
z

File: C++/atm.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

// Fungsi untuk format angka jadi rupiah (pakai titik)
string formatRupiah(long long angka) {
    string s = to_string(angka);
    string hasil = "";
    int count = 0;

    // Loop dari belakang, tambahkan titik setiap 3 digit
    for (int i = s.length() - 1; i >= 0; i--) {
        hasil += s[i];
        count++;
        if (count == 3 && i != 0) {
            hasil += ".";
            count = 0;
        }
    }

    reverse(hasil.begin(), hasil.end()); // Balik string biar benar
    return "Rp " + hasil;
}

int main() {
    string nama, valid;
    long long saldo, jumlah;

    // Validasi Nama
    do {
        cout << "Masukkan Nama: ";
        getline(cin, nama);
        cout << "Apakah nama \"" << nama << "\" benar? (TRUE/FALSE): ";
        getline(cin, valid);
    } while (valid != "TRUE" && valid != "true");

    // Input NIM → Saldo
    cout << "Masukkan NIM tanpa huruf: ";
    cin >> saldo;

    cout << "\nNama: " << nama << endl;
    cout << "Saldo Awal: " << formatRupiah(saldo) << endl;

    int pilihan;
    while (true) {
        cout << "\n===== MENU ATM =====" << endl;
        cout << "1. Cek Saldo" << endl;
        cout << "2. Tarik Tunai" << endl;
        cout << "3. Setor Tunai" << endl;
        cout << "4. Transfer" << endl;
        cout << "5. Keluar" << endl;
        cout << "Pilih menu: ";
        cin >> pilihan;

        switch (pilihan) {
            case 1:
                cout << "Saldo Anda: " << formatRupiah(saldo) << endl;
                break;

            case 2:
                cout << "Masukkan jumlah tarik: ";
                cin >> jumlah;
                if (jumlah > saldo) {
                    cout << "Saldo tidak cukup!" << endl;
                } else {
                    saldo -= jumlah;
                    cout << "Tarik tunai berhasil. Saldo sekarang: " << formatRupiah(saldo) << endl;
                }
                break;

            case 3:
                cout << "Masukkan jumlah setor: ";
                cin >> jumlah;
                saldo += jumlah;
                cout << "Setor tunai berhasil. Saldo sekarang: " << formatRupiah(saldo) << endl;
                break;

            case 4:
                cout << "Masukkan jumlah transfer: ";
                cin >> jumlah;
                if (jumlah > saldo) {
                    cout << "Saldo tidak mencukupi!" << endl;
                } else {
                    saldo -= jumlah;
                    cout << "Transfer berhasil. Saldo sekarang: " << formatRupiah(saldo) << endl;
                }
                break;

            case 5:
                cout << "Terima kasih telah menggunakan ATM!" << endl;
                return 0;

            default:
                cout << "Pilihan tidak valid!" << endl;
        }
    }
}
